### DISCUSSION TITLE
onDuplicateKeyUpdate 'where' is not used

### DISCUSSION THREAD

**Discussion by Distractic** [komapper/komapper#942] [Q&A] [answered]
I have this request : 
```kt
val invite = GuildInvite(
            GuildMemberIds(guildId, entityId),
            Instant.EPOCH,
            expiredAt
        )

val meta = _GuildInvite.guildInvite
val query = QueryDsl.insert(meta)
    .onDuplicateKeyUpdate()
    .where {
        meta.expiredAt notEq invite.expiredAt
    }
    .set {
        it.expiredAt eq invite.expiredAt
    }
    .single(invite)
```

However, when executed, the SQL request is : 
`insert into guild_invite as t0_ (guild_id, entity_id, created_at, expired_at) values ($1, $2, $3, $4) on conflict (guild_id, entity_id) do update set expired_at = $5`

How can I set the `expiredAt` only if the value in the database is different than new one?

**Comment by nakamura-to** [comment, marked as answer]
```kotlin
.where {
    meta.expiredAt notEq invite.expiredAt
}
```

Perhaps ` invite.expiredAt` is `null` and therefore the WHERE clause is not generated.
See also https://www.komapper.org/docs/reference/query/querydsl/expression/#comparison-operator .

Try the following code:

```kotlin
.where {
    if (invite.expiredAt == null) {
        meta.expiredAt.isNotNull()
    } else {
       meta.expiredAt notEq invite.expiredAt
    }
}
```
